### ISSUE TITLE
Return refresh token from login

### ISSUE THREAD

**Issue by Vichy97** [spotify/android-auth#47]
When I login and select "keep me logged in" I get back an access token with an "expires in" field set to 3600. What happens when this time is up? Are they logged out? Can we get a refresh token to keep using that access token without needing to login again?

**Comment by wetracyk44** [comment]
.com

**Comment by JanStoltman** [comment]
+1
It looks like the webview keeps user logged in for some time and on next login attempt returns new access token immediately. It would be much more useful if we could get a refresh token to avoid opening webview again.

**Comment by maracuja-juice** [comment]
You need to have a webservice for that. See their tutorial on their website.

**Comment by JanStoltman** [comment]
@maracuja-juice Do you mean a webview? But still, having to open a webview whenever user opens an application is not as convenient as a solution with a refresh token could be.

**Comment by maracuja-juice** [comment]
No I mean a webservice. See the "Client Credentials Flow" for that: https://developer.spotify.com/documentation/general/guides/authorization-guide/#client-credentials-flow

**Comment by JanStoltman** [comment]
@maracuja-juice I see now, there are three separate auth flows and this library supports only one of those, am I right? Thanks for clarification!

**Comment by maracuja-juice** [comment]
@JanStoltman That's correct. Because with the flow for the refresh token you need to use the secret key, and you **really shouldn't** store this on a users device.
